File: 品味Java的21種設計模式/dp_cpp/chainofresponsibility/example6/SaleModel.cpp
#include "SaleModel.h"

namespace cn {
namespace javass {
namespace dp {
namespace chainofresponsibility {
namespace example6 {

std::string SaleModel::getGoods() {
    return goods;
}

void SaleModel::setGoods(std::string goods) {
    this->goods = goods;
}

int SaleModel::getSaleNum() {
    return saleNum;
}

void SaleModel::setSaleNum(int saleNum) {
    this->saleNum = saleNum;
}

std::string SaleModel::ToString() {
    return "商品名称=" + goods + ",销售数量=" + saleNum;
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/decorator/example6/SaleModel.java
package cn.javass.dp.decorator.example6;
/**
 * 封装销售单的数据，简单的示意一些
 */
public class SaleModel {
    /**
     * 销售的商品
     */
    private String goods;
    /**
     * 销售的数量
     */
    private int saleNum;
    public String getGoods() {
        return goods;
    }
    public void setGoods(String goods) {
        this.goods = goods;
    }
    public int getSaleNum() {
        return saleNum;
    }
    public void setSaleNum(int saleNum) {
        this.saleNum = saleNum;
    }

    public String toString() {
        return "商品名称=" + goods + ",购买数量=" + saleNum;
    }
}